### PR TITLE
Fix "show password" icon styles

### LINKED ISSUE #83: Adjust styles of the "Show password" icon
Ref: https://github.com/WordPress/wporg-two-factor/issues/5#issuecomment-1415629776

- [x] The "Show password" icon is better to be inside the input field.
image
- [x] Have a toggled state for this icon.
image
- [x] The focus style here is a bit awkward. (It could be square if the icon sat inside the input)
image
- [x] The input should not be pushed when toggling state.
Screen Capture on 2023-03-17 at 09-00-48

### PR DESCRIPTION
**Pull request by outdoor2kode**
Fixes #83 

This PR fixes 
* When toggled, the "show password" icon would squeeze the input on the left.
* Make the icon inside the input field. 
* The styles were also modified to accommodate the 1password icon.
* Add a toggled state for the show password icon.
  * The `unseen` icon provided in the issue has already been added to [Gutenberg](https://github.com/WordPress/gutenberg/pull/49254), and it's expected to be in production on GB 15.5, probably 2-3 weeks later. I've opened another [branch](https://github.com/WordPress/wporg-two-factor/compare/testing/show-password-icon-style?expand=1) that had `unseen.svg` included so that it can be tested if anyone wants to.
* Run `lint:js`. 
  * It looks like there were many changes in `settings/src/components/password.js`, but most of them were just linting updates.

## Screencasts
https://user-images.githubusercontent.com/18050944/226954384-86416da1-2925-4f3f-9c2e-98ad3421a4fb.mp4

### CODE REVIEW

File: settings/package.json
Comment (iandunn) [inline settings/package.json]: [Core already includes this](https://github.com/WordPress/WordPress/blob/1b393fbb15de13089b5c97a9c3975b642fbfbf7b/wp-includes/script-loader.php#L111), so bundling another version in our JS build might cause conflicts, and unnecessarily make the build larger.

Is there a tangible reason to add it, like the `import` statements breaking in some environments? Or was it added to document that it's used?

If it's the latter case, would it be better to mark it as being an global/external or something? IIRC there's a way to do that. 

We could also move it to `devDependencies` if it's needed for a tangible reason, though.

File: settings/src/components/password.js
Comment (iandunn) [inline settings/src/components/password.js:47]: The linting/spelling changes are good, thanks! Ideally they'll be in a separate commit from the others (in `trunk`, not just this branch), so it's easier to see distinguish logic changes from linting. 

That's not a big deal though, so don't worry if the rebasing would take too long etc.

File: settings/src/components/password.js
Comment (iandunn) [inline settings/src/components/password.js:26]: Great job adding the icon to Gutenberg!

I guess we'll need to wait until that's in a stable release to merge this?

File: settings/src/components/password.scss
Comment (iandunn) [inline settings/src/components/password.scss:12]:
```suggestion
			/* This includes room for the `(un)seen` icon, and also for the icon that 1password etc add to password fields. */
			padding-right: 35px;
```

**Review by iandunn — APPROVED** [review]
I left a few small comments, but this looks really good. Great job!

File: settings/package.json
Comment (outdoor2kode) [reply settings/package.json]: > Or was it added to document that it's used?

Yeah, it was added because of linting error.

`10:1 error 'lodash' should be listed in the project's dependencies. Run 'npm i -S lodash' to add it  import/no-extraneous-dependencies`

> IIRC there's a way to do that.

Perhaps something like this [one](https://github.com/WordPress/wporg-developer/blob/trunk/source/wp-content/themes/wporg-developer-2023/js/function-reference.js#L1)? 

Unfortunately, adding that comment doesn't fix the error, I've checked how Gutenberg handled it, and they just put it to `devDependencies`([ref](https://github.com/WordPress/gutenberg/blob/trunk/package.json#L205)) (The version is different from the core though, and they seem to [manually](https://github.com/WordPress/gutenberg/commit/d37bcb48d7028aa7701ff129ad6e338c65d1666f) updated it)

I've tried moving it to `devDependencies` and it solved the linting error, but I guess it's ok not to add it.

File: settings/src/components/password.js
Comment (outdoor2kode) [reply settings/src/components/password.js:47]: Yeah, you're right. Will do it that way!

File: settings/src/components/password.js
Comment (outdoor2kode) [reply settings/src/components/password.js:26]: If it is expected to be deployed soon after merging it, then it should wait until the GB 15.5 is released to be merged. Otherwise, it would cause errors.

Comment (outdoor2kode) [reply settings/src/components/password.js:26]: Label: `[Status] On Hold` was added. I'll keep track of the Guteberg release schedule.

File: settings/package.json
Comment (iandunn) [reply settings/package.json]: Ah, fixing the linting error makes sense 👍🏻 

It sounds like [webpack externals](https://webpack.js.org/configuration/externals/) might be a good way to do it. `wp-scripts` provides a way to [extend the default config](https://github.com/WordPress/gutenberg/tree/trunk/packages/scripts#provide-your-own-webpack-config). I'm not sure if the linter would pick up on that or not, though.

I think leaving it in `devDependencies` is a good solution too though 👍🏻

File: settings/src/components/password.js
Comment (iandunn) [reply settings/src/components/password.js:26]: That sounds good 👍🏻  It is usually best to deploy soon after merging.

File: settings/package.json
Comment (StevenDufresne) [reply settings/package.json]: I agree with the `devDependencies` approach.

**Review by StevenDufresne — APPROVED** [review]

**Pull request merged**
